src/callbacks/logging/checkpointer.jl

```diff
@@ -25,7 +29,34 @@ function on(::EpochEnd, phase::AbstractTrainingPhase, checkpointer::Checkpointer
     loss = last(learner.cbstate.metricsepoch[phase], :Loss)[2]
     epoch = learner.cbstate.history[phase].epochs
     filename = "checkpoint_epoch_$(lpad(string(epoch), 3, '0'))_loss_$loss.bson"
+
     savemodel(learner.model, joinpath(checkpointer.folder, filename))
+
+    if !isnothing(checkpointer.keep_top_k)
+        process_top_k_checkpoints(checkpointer, filename, loss)
+    end
+end
+
+try_fs_remove(path::String) =
+    try; Base.Filesystem.rm(path)
+    catch e; @warn e; end
```

Comment:
I would recommend providing a better warning message (since `@warn` doesn't have a stacktrace) or using `rm(..., force=true)`.

Reply:
Thanks. I went with `rm(...; force=true)`. Mostly I wanted to make sure the checkpointer doesn't crash the training hours in.